src/hazelcore/Builtin.re

```diff
@@ -0,0 +1,113 @@
+open Sexplib.Std;
+
+[@deriving sexp]
+type args = list(DHExp.t);
+
+[@deriving sexp]
+type eval = (args, DHExp.t => EvaluatorResult.t) => EvaluatorResult.t;
+
+[@deriving sexp]
+type elab = DHExp.t;
+
+[@deriving sexp]
+type t = {
+  ident: Var.t,
+  ty: HTyp.t,
+  eval,
+  elab,
+};
+
+module Impl = {
+  /* Evaluator alias. */
+  [@deriving sexp]
+  type evaluate = DHExp.t => EvaluatorResult.t;
```

Comment:
this is used in the type `eval` above, maybe move this out to the top level / interface.

everything else in Impl you can also move out and just leave out of the rei interface to avoid it being exported